torchvision/models/detection/transform.py

```diff
@@ -76,17 +89,17 @@ def torch_choice(self, l):
     def resize(self, image, target):
         # type: (Tensor, Optional[Dict[str, Tensor]])
         h, w = image.shape[-2:]
-        im_shape = torch.tensor(image.shape[-2:])
-        min_size = float(torch.min(im_shape))
-        max_size = float(torch.max(im_shape))
         if self.training:
             size = float(self.torch_choice(self.min_size))
         else:
             # FIXME assume for now that testing uses the largest scale
             size = float(self.min_size[-1])
-        scale_factor = size / min_size
-        if max_size * scale_factor > self.max_size:
-            scale_factor = self.max_size / max_size
+        if torchvision._is_tracing():
+            from torch.onnx import operators
+            im_shape = operators.shape_as_tensor(image)[-2:]
```

Comment:
@fmassa Please review this update. Since most of the code is similar in two paths, I updated the branch as such.